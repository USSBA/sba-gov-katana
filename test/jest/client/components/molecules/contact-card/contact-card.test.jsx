/*global expect*/

import React from 'react'
import ContactCard from 'client/components/molecules/contact-card/contact-card.jsx'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import _ from 'lodash'

const card = {
  type: 'contact',
  cdcNumber: '12345',
  city: 'Galactic City',
  contactName: null,
  email: 'tarkin@deathstar.org',
  firsNumber: '1234',
  link: 'http://starwars.wikia.com/wiki/Wilhuff_Tarkin',
  phoneNumber: '123-456-5665',
  state: 'MD',
  stateServed: 'Maryland',
  streetAddress: '123 Death Star Street, Suite 1',
  category: 'CDC/504',
  zipCode: 12345,
  title: 'Grand Moff Tarkin',
  id: 2974
}

describe('ContactCard', () => {
  test('should render with all the information', () => {
    let cardData = _.clone(card)
    const component = renderer.create(<ContactCard {...cardData} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render without an address ', () => {
    let cardData = _.omit(card, 'streetAddress')
    const component = shallow(<ContactCard {...cardData} />)
    expect(component.find('.contact-card').children()).toHaveLength(4)
  })

  test('should render without a zip code ', () => {
    let cardData = _.omit(card, 'zipCode')
    const component = shallow(<ContactCard {...cardData} />)
    expect(component.find('.contact-card').children()).toHaveLength(4)
  })

  test('should render without a phone number ', () => {
    let cardData = _.omit(card, 'phoneNumber')
    const component = shallow(<ContactCard {...cardData} />)
    expect(component.find('.contact-card').children()).toHaveLength(4)
  })

  test('should render without a website link ', () => {
    let cardData = _.omit(card, 'link')
    const component = shallow(<ContactCard {...cardData} />)
    expect(component.find('.contact-card').children()).toHaveLength(4)
  })

  test('should render without a website link if link is an empty object', () => {
    let cardData = _.clone(card)
    _.set(cardData, 'link', {})
    const component = shallow(<ContactCard {...cardData} />)
    expect(component.find('.contact-card').children()).toHaveLength(4)
  })

  test('should render without a website link if link is an empty string', () => {
    let cardData = _.clone(card)
    _.set(cardData, 'link', '')
    const component = shallow(<ContactCard {...cardData} />)
    expect(component.find('.contact-card').children()).toHaveLength(4)
  })

  test('should render without a website link if link is null', () => {
    let cardData = _.clone(card)
    _.set(cardData, 'link', null)
    const component = shallow(<ContactCard {...cardData} />)
    expect(component.find('.contact-card').children()).toHaveLength(4)
  })

  test('should render without the state served (which is normally invisible)', () => {
    let cardData = _.omit(card, 'stateServed')
    const component = shallow(<ContactCard {...cardData} />)
    expect(component.find('.contact-card').children()).toHaveLength(5)
  })
})
