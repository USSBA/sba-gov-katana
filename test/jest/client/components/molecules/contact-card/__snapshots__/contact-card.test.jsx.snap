// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactCard should render with all the information 1`] = `
<div
  className="contact-card box contactCard"
  data-testid="contact-card"
>
  <h6
    data-cy="contact card title"
  >
    Grand Moff Tarkin
  </h6>
  <div
    className="row"
    data-cy="contact address"
  >
    <div
      className="firstColumn"
    >
      <i
        alt="123 Death Star Street, Suite 1<br />Galactic City, MD 12345"
        aria-hidden="true"
        className="fa fa-map-marker"
      />
    </div>
    <div
      className="secondColumn"
    >
      <a
        dangerouslySetInnerHTML={
          Object {
            "__html": "123 Death Star Street, Suite 1<br />Galactic City, MD 12345",
          }
        }
        href="https://maps.google.com?q=123%20Death%20Star%20Street%2C%20Suite%201%0AGalactic%20City%2C%20MD%2012345"
      />
    </div>
  </div>
  <div
    className="row"
    data-cy="contact phone"
  >
    <div
      className="firstColumn"
    >
      <i
        alt="123-456-5665"
        aria-hidden="true"
        className="fa fa-phone"
      />
    </div>
    <div
      className="secondColumn"
    >
      <a
        dangerouslySetInnerHTML={
          Object {
            "__html": "123-456-5665",
          }
        }
        onClick={[Function]}
        style={Object {}}
        target="_self"
      />
    </div>
  </div>
  <div
    className="row"
    data-cy="contact link"
  >
    <div
      className="firstColumn"
    >
      <i
        alt="Visit website"
        aria-hidden="true"
        className="fa fa-external-link-square"
      />
    </div>
    <div
      className="secondColumn"
    >
      <a
        dangerouslySetInnerHTML={
          Object {
            "__html": "Visit website",
          }
        }
        href="http://starwars.wikia.com/wiki/Wilhuff_Tarkin"
      />
    </div>
  </div>
  <div
    className="row"
    data-cy="contact email"
  >
    <div
      className="firstColumn"
    >
      <i
        alt="tarkin@deathstar.org"
        aria-hidden="true"
        className="fa fa-envelope-o"
      />
    </div>
    <div
      className="secondColumn"
    >
      <a
        dangerouslySetInnerHTML={
          Object {
            "__html": "tarkin@deathstar.org",
          }
        }
        onClick={[Function]}
        style={Object {}}
        target="_self"
      />
    </div>
  </div>
</div>
`;
